packages/vue/src/components/atoms/SfImage/SfImage.js

```diff
@@ -64,20 +64,43 @@ export default {
   methods: {
     hoverHandler(state) {
       this.overlay = state;
+    },
+
+    initLozad: function() {
+      const vm = this;
+      this.$nextTick(() => {
+        const observer = lozad(vm.$refs.imgLazy, {
+          loaded: function() {
+            vm.loaded = true;
+          }
+        });
+        observer.observe();
+      });
     }
   },
 
   mounted() {
     if (this.lazy !== false) {
-      const vm = this;
-      const observer = lozad(vm.$refs.imgLazy, {
-        loaded: function() {
-          vm.loaded = true;
-        }
-      });
-      observer.observe();
+      this.initLozad();
     } else {
       this.loaded = true;
     }
+  },
+
+  watch: {
+    lazy: function(newValue, oldValue) {
+      // init lozad if lazy loading was previously disabled
+      if (!oldValue && newValue) {
+        this.initLozad();
+      }
+      // if lazy loading was previously enabled, remove lozad classes and
+      // remove spurious img tag added by lozad if src is a multi-size object
+      if (oldValue && !newValue) {
+        this.$refs.imgLazy.removeAttribute("data-loaded");
+        if (this.$refs.imgLazy.tagName === "PICTURE") {
+          this.$refs.imgLazy.querySelector("img").remove();
+        }
+      }
+    }
```

Comment:
Not sure yet, but I think on second case we can just remove `imgLazy` element completely, would be a simpler solution, wdyt?

```js
lazy (newValue, oldValue) {
  // init lozad if lazy loading was previously disabled
  if (!oldValue) {
    if (newValue) {
      this.initLozad();
    }
  } else if (!newValue) {
    // if lazy loading was previously enabled, remove lozad classes and
    // remove spurious img tag added by lozad if src is a multi-size object
    const $lozadImg = this.$refs.imgLazy
    if ($lozadImg) {
      $lozadImg.remove()
    }
  }
}
```

Reply:
I tried that but it didn't work. 😕  The whole `picture` tag is removed from the DOM–as expected–but it doesn't come back^^ I guess the diffing algorithm doesn't expect the full tag to be removed but just patched the changed parts, but as we removed the whole thing, there is nothing to be diffed/changed/added anymore. Even after triggering the lazy state again (off, on, off, on, …): the `picture` tag doesn't get rendered anymore in this case. 🤷‍♂

Reply:
ok @EduardDopler , no problem as you fixed the bug, thank you :grin: